objective-c/src/Ice/PropertiesI.mm

```diff
@@ -270,21 +270,21 @@ -(void) load:(NSString*)file
 
     id<ICEPropertiesAdminUpdateCallback> _callback;
 };
-typedef IceUtil::Handle<UpdateCallbackI> UpdateCallbackIPtr;
+using UpdateCallbackIPtr = Ice::SharedPtr<UpdateCallbackI>;
 
 }
 
 @implementation ICEPropertiesAdminUpdateCallback
 @end
 
-#define NATIVEPROPERTIESADMIN dynamic_cast<Ice::NativePropertiesAdmin*>(object_)
+#define NATIVEPROPERTIESADMIN std::dynamic_pointer_cast<Ice::NativePropertiesAdmin>(object_)
```

Comment:
I think we should remove this macro, which its only purpose seems to be to save some typing.

Reply:
I agree, for a future PR. Note that we have several of these macros.